server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -102,17 +101,19 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
         }
     }
 
-    private Stream<MailboxId> retrieveMailboxIds(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
+    private Stream<Optional<Mailbox>> retrieveMailboxes(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
```

Comment:
private method is not an excuse to be lazy (:
if you don't care about empty element, you have to return a Stream<Mailbox>, doing otherwise is just promoting easy writing over easy reading